cmd/proxy/main.go

```diff
@@ -0,0 +1,173 @@
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"os"
+	"regexp"
+
+	"k8s.io/kube-openapi/pkg/common"
+	"k8s.io/kube-openapi/pkg/handler"
+	"k8s.io/kube-openapi/pkg/handler3"
+	"k8s.io/kube-openapi/pkg/spec3"
+	"k8s.io/kube-openapi/pkg/validation/spec"
+)
+
+func serve_v2(swagger spec.Swagger, h common.PathHandler) error {
+	o, err := handler.NewOpenAPIService(&swagger)
+	if err != nil {
+		return fmt.Errorf("failed to create OpenAPI v2 service: %v", err)
+	}
+	err = o.RegisterOpenAPIVersionedService("/openapi/v2", h)
+	if err != nil {
+		return fmt.Errorf("failed to register OpenAPI service: %v", err)
+	}
+	return nil
+}
+
+func serve_v3(v3 map[XKubernetesGroupVersion]spec3.OpenAPI, h common.PathHandler) error {
+	o, err := handler3.NewOpenAPIService(nil)
+	if err != nil {
+		return fmt.Errorf("failed to create OpenAPI v3 service: %v", err)
+	}
+
+	h.Handle("/openapi/v3", http.HandlerFunc(o.HandleDiscovery))
```

Comment:
whats wrong with https://github.com/kubernetes/kube-openapi/blob/352e363eb79bf637a9a7aa5e1db7c67215fd0414/pkg/handler3/handler.go#L260-L264

Reply:
it's uses a "handler" interface that implements the `HandlePrefix` function, which exists in k8s, but not in kube-openapi. I don't really need/want to import something more complicated than the old `http/net` mux.